discovery-infra/tests/base_test.py

```diff
@@ -341,7 +343,41 @@ def kube_api_client(self):
 
     @pytest.fixture()
     def kube_api_context(self, kube_api_client):
-        kube_api_context = KubeAPIContext(kube_api_client, clean_on_exit=global_variables.test_teardown)
+        kube_api_context = KubeAPIContext(kube_api_client)
 
         with kube_api_context:
             yield kube_api_context
+
+    @pytest.fixture(scope="function")
+    def update_olm_config(self) -> Callable:
+        def update_config(config, operators=None):
+            if operators is None:
+                operators = operators_utils.parse_olm_operators_from_env()
+
+            config.worker_memory = operators_utils.resource_param(config.worker_memory, consts.OperatorResource.WORKER_MEMORY_KEY, operators)
+            config.master_memory = operators_utils.resource_param(config.master_memory, consts.OperatorResource.MASTER_MEMORY_KEY, operators)
+            config.worker_vcpu = operators_utils.resource_param(config.worker_vcpu, consts.OperatorResource.WORKER_VCPU_KEY, operators)
+            config.master_vcpu = operators_utils.resource_param(config.master_vcpu, consts.OperatorResource.MASTER_VCPU_KEY, operators)
+            config.workers_count = operators_utils.resource_param(config.workers_count, consts.OperatorResource.WORKER_COUNT_KEY, operators)
+            config.worker_disk = operators_utils.resource_param(config.worker_disk, consts.OperatorResource.WORKER_DISK_KEY, operators)
+            config.master_disk = operators_utils.resource_param(config.master_disk, consts.OperatorResource.MASTER_DISK_KEY, operators)
+            config.master_disk_count = operators_utils.resource_param(config.master_disk_count, consts.OperatorResource.MASTER_DISK_COUNT_KEY, operators)
+            config.worker_disk_count = operators_utils.resource_param(config.worker_disk_count, consts.OperatorResource.WORKER_DISK_COUNT_KEY, operators)
+
+            return config
+        
+        yield update_config
+
+    @pytest.fixture(scope="function")
+    def update_cluster_config(self) -> Callable:
+        def update_config(config, operators=None):
+            if operators is None:
+                operators = operators_utils.parse_olm_operators_from_env()
+            
+            config.workers_count = operators_utils.resource_param(config.workers_count, consts.OperatorResource.WORKER_COUNT_KEY, operators)
+            config.nodes_count = config.masters_count + config.workers_count
+            config.olm_operators = [operators]
+            
+            return config
+        
+        yield update_config
```

Comment:
Merge those two to get 2 configs and update them both instead

Reply:
My thinking was that we both config may require to be updated in different places but if you think we should be good with one I am happy to update.